src/rai/rai/tools/ros/native.py

```diff
@@ -223,6 +225,57 @@ def _run(self, topic_name: str):
             return str(msg), {}
 
 
+class Ros2GenericServiceCallerInput(BaseModel):
+    service_name: str = Field(..., description="Name of the ROS2 service to call")
+    service_type: str = Field(
+        ..., description="Type of the ROS2 service in typical ros2 format"
+    )
+    request_args: Dict[str, Any] = Field(
+        ..., description="Arguments for the service request"
+    )
+
+
+class Ros2GenericServiceCaller(Ros2BaseTool):
```

Comment:
```suggestion
class Ros2GenericServiceCaller(Ros2AsyncBaseTool):
```

Reply:
No such thing as Ros2AsyncBaseTool

Reply:
Sorry, I meant `Ros2BaseActionTool` - but let's skip it for now.